lib/rouge/lexers/bibtex.rb

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+# regex based on https://github.com/SaswatPadhi/prismjs-bibtex and https://github.com/alecthomas/chroma/blob/master/lexers/b/bibtex.go
+
+module Rouge
+  module Lexers
+    class BibTeX < RegexLexer
+      title 'BibTeX'
+      desc "BibTeX"
+      tag 'bibtex'
+      filenames '*.bib'
+
+      valid_punctuation = /@\!\$\&\*\+\-\.\/\:\;\<\>\?\[\\\]\^\`\|\~/
+      valid_name = /[a-z\_#{valid_punctuation}][\w#{valid_punctuation}]*/io
+
+      state :root do
+        mixin :whitespace
+        rule %r/@comment/i, Comment
+        rule %r/@preamble/i do
+          token Name::Class
+          push :closing_brace
+          push :value
+          push :opening_brace
+        end
+        rule %r/@string/i do
+          token Name::Class
+          push :closing_brace
+          push :field
+          push :opening_brace
+        end
+        rule %r/@#{valid_name}/o do
+          token Name::Class
+          push :closing_brace
+          push :command_body
+          push :opening_brace
+        end
+        rule %r/.+/, Comment
+      end
+
+      state :opening_brace do
+        mixin :whitespace
+        rule %r/[{(]/, Punctuation, :pop!
+      end
+
+      state :closing_brace do
+        mixin :whitespace
+        rule %r/[})]/, Punctuation, :pop!
+      end
+
+      state :command_body do
+        mixin :whitespace
+        rule %r/[^\s\,\}]+/ do
+          token Name::Label
+          pop!
+          push :fields
+        end
+      end
+
+      state :fields do
+        mixin :whitespace
+        rule %r/,/, Punctuation, :field
+        rule(//) { pop! }
+      end
+
+      state :field do
+        mixin :whitespace
+        rule(/#{valid_name}/o) { token Name::Attribute; push :value; push :equal_sign }
+        rule(//) { pop! }
+      end
+
+      state :equal_sign do
+        mixin :whitespace
+        rule %r/=/, Punctuation, :pop!
+      end
+
+      state :value do
+        mixin :whitespace
+        rule %r/#{valid_name}/o, Name::Variable
+        rule %r/"/, Literal::String, :quoted_string
+        rule %r/\{/, Literal::String, :braced_string
+        rule %r/[\d]+/, Literal::Number
```

Comment:
Do numbers just have digits? There's no `.` or `,` permitted?

Reply:
See [my comment below](https://github.com/rouge-ruby/rouge/pull/1360#issuecomment-631400145).

Reply:
A fuller explanation is below but the short answer is no.